.github/workflows/pull_request.yml

```diff
@@ -106,6 +106,9 @@ jobs:
       - name: Running schema tests
         run: ./scripts/validate_test_schemas.sh
   test-functional:
+    strategy:
+      matrix:
+        suite: [ timeout_modal, features, general, components ]
     needs: [python-dependencies, node-dependencies]
```

Comment:
Why does test-functional need python dependencies ?

Reply:
Cause the generates pages are created by `generate_pages.py`. One day it won't be Py 😄

Reply:
can we not generate it as `artifacts` in previous steps and share with following functionals tests . We dont need dependencies

Reply:
We definitely could, but personally don't think that should block this though, as that is an improvement external to the scope of this. Also, the python dependencies are cached, I don't believe they get reinstalled at this point. But ya, we can look to do further clean up.

Reply:
it could save 2 mins of docker-compose , probably need a card to do that so this can go in . We need this urgently.

Reply:
As discussed, these deps are for actions and not docker. But we can still look to improve from a best practice point of view separately. Once we move generating the pages out into artifacts, this can go.